### ISSUE TITLE
NullPointerException when creating notification

### ISSUE THREAD

**Issue by nzayatz14** [aws-amplify/aws-sdk-android#2357]
**Describe the bug**

Occasionally seeing NPEs in my Google Play Console Crashes & ANRs section

```
java.lang.NullPointerException: 
  at com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.createNotification (NotificationClientBase.java:593)
  at com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.access$100 (NotificationClientBase.java:59)
  at com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase$1.run (NotificationClientBase.java:732)
  at java.lang.Thread.run (Thread.java:919)
```

**Which AWS service(s) are affected?**

AWS-Pinpoint

**Environment Information (please complete the following information):**

Recently Updated AWS SDK packages

From:
```
implementation 'com.amazonaws:aws-android-sdk-pinpoint:2.15.2'
implementation('com.amazonaws:aws-android-sdk-mobile-client:2.15.2@aar') { transitive = true }
```

To:
```
implementation 'com.amazonaws:aws-android-sdk-pinpoint:2.22.0'
implementation('com.amazonaws:aws-android-sdk-mobile-client:2.22.0@aar') { transitive = true }
```

**Comment by raphkim** [comment]
Fix for this issue was released in v2.29.0. Please try updating to the latest version (v2.30.0 as of writing) to see if your issue is resolved.

**Comment by nzayatz14** [comment]
@raphkim I just recently started getting this issue again on version 2.37.1
